tests/device_selection/test_predictor_ml.py

```diff
@@ -5,145 +5,95 @@
-    """
-    predictor = ml.Predictor()
-    assert predictor.clf is None
-    predictor.train_random_forest_classifier(visualize_results=False, save_classifier=False)
-
-    assert predictor.clf is not None
-
-    qc = benchmark_generator.get_benchmark("ghz", 1, 3)
-    prediction = predictor.predict_probs(qc, "expected_fidelity")
-    for elem in prediction:
-        assert 0 <= elem <= 1
-
-    file = Path("test_qasm.qasm")
-    qc = benchmark_generator.get_benchmark("dj", 1, 3)
-    with file.open("w", encoding="utf-8") as f:
-        dump(qc, f)
-
-    prediction = predictor.predict_probs(file, "expected_fidelity")
-    for elem in prediction:
-        assert 0 <= elem <= 1
-
-    with pytest.raises(
-        FileNotFoundError, match=re.escape("The ML model is not trained yet. Please train the model before using it.")
-    ):
-        ml.helper.predict_device_for_figure_of_merit(qc, "false_input")  # type: ignore[arg-type]
-
-    (ml.helper.get_path_trained_model("expected_fidelity").parent / "non_zero_indices_expected_fidelity.npy").unlink()
-
-
-def test_performance_measures() -> None:
-    """Test the calculation of the performance measures for a given set of scores and labels."""
-    predictor = ml.Predictor()
-    figure_of_merit: Literal["expected_fidelity"] = "expected_fidelity"
-
-    training_data = predictor.get_prepared_training_data(figure_of_merit=figure_of_merit, save_non_zero_indices=False)
-
-    y_test = training_data.y_test
-    indices_test = training_data.indices_test
-    names_list = training_data.names_list
-    scores_list = training_data.scores_list
-
-    assert len(y_test) > 0
-    assert len(indices_test) > 0
-    assert len(names_list) > 0
-    assert len(scores_list) > 0
-
-    scores_filtered = [scores_list[i] for i in indices_test]
-    names_filtered = [names_list[i] for i in indices_test]
-
-    # Test calc_performance_measures
-    res, relative_scores = predictor.calc_performance_measures(scores_filtered, y_test, y_test)
-    assert all(res)
-    assert not any(relative_scores)
-
-    # Test generate_eval_histogram
-    predictor.generate_eval_histogram(res, show_plot=False)
-    histogram_path = Path("results/histogram.pdf")
-    assert histogram_path.is_file(), "File does not exist"
-    histogram_path.unlink()
-
-    # Test generate_eval_all_datapoints
-    predictor.generate_eval_all_datapoints(names_filtered, scores_filtered, y_test, y_test)
-    result_path = Path("results/y_pred_eval_normed.pdf")
-    assert result_path.is_file(), "File does not exist"
-    result_path.unlink()
-
-
-def test_compile_all_circuits_for_dev_and_fom() -> None:
-    """Test the compilation of all circuits for a given device and figure of merit."""
-    predictor = ml.Predictor()
+def test_entire_setup() -> None:
+    """Test the training of a random forest classifier."""
+    # create the ML Predictor
+    predictor = ml.Predictor(devices=["ionq_harmony"], figure_of_merit="expected_fidelity")
```

Comment:
It's a bit of a boring and somewhat useless test to train an ML classifier for exactly one device. Might be good to choose at least two here so that this actually tests something proper.

Reply:
That indeed is true. I was just wondering since more devices do not increase coverage but increase the CI runtime much since the RL training takes the majority of time. Do you think it is worth to increase the runtime by 1.5-2x?

Reply:
If you choose another small device where compilation is fast, this should still be reasonable, right?

Ideally, it would be possible to just define a small MockDevice right here in the predictor. This might only be possible after a bit of additional refactor.

Reply:
Actually, even for small devices, the training takes rather long due to, e.g., the setup time when executing a BQSKit compilation pass. Therefore, I would prefer to keep it as is for now since the runtime is already longer than one hour due to not having the individually test runs anymore.

Reply:
Fine with me 👍🏻